pegasus/helpers/section_api_helpers.rb

```diff
@@ -235,22 +258,25 @@ def self.valid_scripts(user_id = nil)
         where(where_clause).
         select(:id, :name, :hidden).
         all.
-        map do |script|
-          name = ScriptConstants.teacher_dashboard_name(script[:name])
-          first_category = ScriptConstants.categories(script[:name])[0] || 'other'
-          position = ScriptConstants.position_in_category(name, first_category)
-          category_priority = ScriptConstants.category_priority(first_category)
-          name = I18n.t("#{name}_name", default: name)
-          name += " *" if script[:hidden]
-          {
-            id: script[:id],
-            name: name,
-            script_name: script[:name],
-            category: I18n.t("#{first_category}_category_name", default: first_category),
-            position: position,
-            category_priority: category_priority
-          }
-        end
+        map {|script| assignable_info(script, script[:hidden])}
+  end
+
+  @@course_cache = {}
+  # Mimic the behavior of valid_scripts, but return courses instead. Also simpler
+  # in that we don't have to worry about hidden courses.
+  def self.valid_courses
+    course_cache_key = I18n.locale.to_s
+
+    return @@course_cache[course_cache_key] if @@course_cache.key?(course_cache_key)
+
+    return {} unless (Dashboard.db[:courses].count rescue nil)
```

Comment:
I'm not entirely sure. I copied this from the valid_scripts, and assumed it was there for some good reason.